packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -83,6 +103,88 @@ class StorageS3Service {
     }
   }
 
+  /// Takes in input from [AmplifyStorageS3Dart.getProperties] to compose a
+  /// [HeadObjectRequest] and send to S3 service, then returns a
+  /// [S3StorageGetPropertiesResult] based on [HeadObjectOutput] returned by
+  /// [S3Client.headObject] API.
+  ///
+  /// {@macro storage.s3_service.throw_exception_unknown_smithy_exception}
+  Future<S3StorageGetPropertiesResult> getProperties({
+    required String key,
+    required S3StorageGetPropertiesOptions options,
+  }) async {
+    final resolvedPrefix = await _getResolvedPrefix(
+      storageAccessLevel: options.storageAccessLevel,
+      identityId: options.targetIdentityId,
+    );
+
+    final keyToGetProperties = '$resolvedPrefix$key';
+
+    final request = HeadObjectRequest.build((builder) {
+      builder
+        ..bucket = _defaultBucket
+        ..key = keyToGetProperties;
+    });
+
+    try {
+      return S3StorageGetPropertiesResult.fromHeadObjectOutput(
+        await _defaultS3Client.headObject(request),
+        key: key,
+      );
+    } on UnknownSmithyHttpException catch (error) {
+      throw S3StorageException.fromUnknownSmithyHttpException(error);
+    }
+  }
+
+  /// Takes in input from [AmplifyStorageS3Dart.getUrl] to create a
+  /// `GET` [AWSHttpRequest], then `presign` it with [AWSSigV4Signer], and
+  /// returns a [S3StorageGetUrlResult] containing the presigned [Uri].
+  ///
+  /// {@macro storage.s3_service.throw_exception_unknown_smithy_exception}
+  Future<S3StorageGetUrlResult> getUrl({
+    required String key,
+    required S3StorageGetUrlOptions options,
+  }) async {
+    if (options.checkObjectExistence) {
+      // make a HeadObject call for checking object existence
+      // it may throw 404 if object doesn't exist in bucket
+      await getProperties(
+        key: key,
+        options: options.targetIdentityId == null
+            ? S3StorageGetPropertiesOptions(
+                storageAccessLevel: options.storageAccessLevel,
+              )
+            : S3StorageGetPropertiesOptions.forIdentity(
+                options.targetIdentityId!,
+              ),
+      );
+    }
+
+    final resolvedPrefix = await _getResolvedPrefix(
+      storageAccessLevel: options.storageAccessLevel,
+      identityId: options.targetIdentityId,
+    );
+    final keyToGetUrl = '$resolvedPrefix$key';
+
+    final host =
+        '$_defaultBucket.${_getS3EndpointHost(region: _defaultRegion)}';
+    final urlRequest = AWSHttpRequest.get(
+      Uri.https(host, keyToGetUrl),
```

Comment:
I think enforcing HTTPS is fine although we should call it out in the docs since that is not enforced by S3 always

Reply:
Ahh yea, I think there is a requirement with sec review enforcing https that should include presigned url, and also enforcing TSL1.2? Will confirm and verify.

Reply:
Sounds good. Yeah, that's fine, let's call it out either way so that people know.